recipe/build.sh

```diff
@@ -23,14 +23,18 @@ if [[ "${target_platform}" == "linux-aarch64" ]]; then
   export CXXFLAGS="$CXXFLAGS -flax-vector-conversions"
 fi
 
+# This should not be here but the one generated by CMake should be used.
+rm -f folly/folly-config.h
+
 mkdir -p _build
 cd _build
 
-if [[ ! -z "${folly_build_ext+x}" && "${folly_build_ext}" == "jemalloc" ]]
-then
-    CMAKE_ARGS="${CMAKE_ARGS} -DENABLE_JEMALLOC=ON"
+if [[ ! -z "${folly_build_ext+x}" && "${folly_build_ext}" == "jemalloc" ]]; then
+    export CXXFLAGS="${CXXFLAGS} -DFOLLY_HAVE_LIBJEMALLOC=1"
```

Comment:
From my limited time and what I can see locally with:

```
git diff v2024.05.06.00...v2024.05.13.00
```

Since [`folly/folly-config.h` (now?) sets](https://github.com/facebook/folly/compare/v2024.05.06.00...v2024.05.13.00#diff-e1dc9a8df267b7367905f234329135f43613a715da66a997674b0240c5ab139bR384-R393)


```cpp
// We depend on JEMalloc headers in fbcode, so use them (note that when using
// sanitizers, `#ifdef` gates in the code will not use JEMalloc headers,
// despite this setting).  This set of defines is a bit complex, but we try to
// come up with a set of defines that will only match in fbcode, where JEMalloc
// is available.
#if FOLLY_HAVE_LIBJEMALLOC && !defined(FOLLY_USE_JEMALLOC) && !FOLLY_MOBILE && \
    !defined(__APPLE__) && !defined(__ANDROID__) && !defined(_WIN32) &&        \
    !defined(__wasm32__)
#define FOLLY_USE_JEMALLOC 1
#endif
```

and:

```cpp
+#if !defined(FOLLY_MOBILE)
+#if defined(__ANDROID__) || \
+    (defined(__APPLE__) &&  \
+     (TARGET_IPHONE_SIMULATOR || TARGET_OS_SIMULATOR || TARGET_OS_IPHONE))
+#define FOLLY_MOBILE 1
+#else
+#define FOLLY_MOBILE 0
+#endif
+#endif // FOLLY_MOBILE
```

Might we need to change our setup to:
```suggestion
    export CXXFLAGS="${CXXFLAGS} -DFOLLY_HAVE_LIBJEMALLOC=1 -DFOLLY_USE_JEMALLOC=1"
```

Reply:
`folly-config.h` has probably been committed to make the build work without CMake. Sadly, it is also less reliable in testing the existence of certain things.

Reply:
Let's try?